pkgs/applications/networking/instant-messengers/vk-messenger/default.nix

```diff
@@ -1,54 +1,74 @@
-{ stdenv, lib, fetchurl, rpmextract, autoPatchelfHook
+{ stdenv, lib, fetchurl, rpmextract, undmg, autoPatchelfHook
 , xorg, gtk3, gnome2, nss, alsaLib, udev, libnotify
 , wrapGAppsHook }:
 
 let
-  version = "5.0.1";
-in stdenv.mkDerivation {
   pname = "vk-messenger";
-  inherit version;
+  version = "5.1.1";
+
   src = {
     i686-linux = fetchurl {
       url = "https://desktop.userapi.com/rpm/master/vk-${version}.i686.rpm";
-      sha256 = "1ji23x13lzbkiqfrrwx1pj6gmms0p58cjmjc0y4g16kqhlxl60v6";
+      sha256 = "1c4pxjg9z7wpir0yc0xh6vca0i58jnzylls2v68fplr0mndc37c1";
     };
     x86_64-linux = fetchurl {
       url = "https://desktop.userapi.com/rpm/master/vk-${version}.x86_64.rpm";
-      sha256 = "01vvmia2qrxvrvavk9hkkyvfg4pg15m01grwb28884vy4nqw400y";
+      sha256 = "0619qsyd499iqfzwi3fx9w5ww8lygl9vq1m4ax9kj09ydcm45r2c";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://desktop.userapi.com/mac/master/vk.dmg";
```

Comment:
We must have a fixed version, not "master"

Reply:
Upstream doesn't provide versioned link for dmg, unlike rpm.

Reply:
@sikmir Then on next dmg reupload hash will become invalid. We don't support sources that get overwritten

Reply:
Of course, that's not good, but there is nothing we can do with it. There are a lot of such cases in nixpkgs, for example https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/medio/default.nix#L12

Reply:
And "master" in the url doesn't mean that dmg will be reuploaded on each "commit", they reupload dmg only on version bump.

Reply:
@sikmir I suppose they just slip away the review. Nix really doesn't support non-fixed schemes.
And actually correct url for medio should be http://dotcolon.net/download/fonts/medio_0200.zip

Reply:
Well, as an alternative way we can use versioned link to web.archive.org, I'm not sure is it OK?

Reply:
@sikmir That should be ok, see https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/babelstone-han/default.nix#L8

Reply:
Updated.